src/lib/sql/sql_translator.cpp

```diff
@@ -1363,9 +1364,61 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_create_table(const hs
 
       column_definition.name = parser_column_definition->name;
       column_definition.nullable = parser_column_definition->nullable;
+
+      // Translate column constraints
+      for (const auto& column_constraint : *parser_column_definition->column_constraints) {
+        if (column_constraint != hsql::ConstraintType::Unique &&
+            column_constraint != hsql::ConstraintType::PrimaryKey) {
+          continue;
+        }
+        const auto constraint_type = column_constraint == hsql::ConstraintType::PrimaryKey
+                                         ? KeyConstraintType::PRIMARY_KEY
+                                         : KeyConstraintType::UNIQUE;
+        table_key_constraints.emplace_back(std::unordered_set<ColumnID>{column_id}, constraint_type);
+        std::cout << "WARNING: "
+                  << std::string{magic_enum::enum_name(constraint_type)} + " constraint for column " +
+                         column_definition.name + " will not be enforced\n";
+      }
+    }
+
+    // Translate table constraints
+    const auto column_count = column_definitions.size();
+    for (const auto& table_constraint : *create_statement.tableConstraints) {
+      std::unordered_set<ColumnID> column_ids;
+      const bool is_primary_key_constraint = table_constraint->type == hsql::ConstraintType::PrimaryKey;
+
+      // Resolve column IDs
+      for (const auto column_name : *table_constraint->columnNames) {
+        for (auto column_id = ColumnID{0}; column_id < column_count; ++column_id) {
+          auto& column_definition = column_definitions[column_id];
+          if (column_definition.name == column_name) {
```

Comment:
Guard clause?

Reply:
Currently, the parser ensures that `columnNames` is not a `nullptr`, however, a (`DebugAssert`) guard does not hurt. `column_count` is set to `column_definitions.size()` and we don't need a guard there.

Reply:
That means what? :)

Reply:
Furthermore, I added a check that all column names defined in a table constraint refer to actual columns. However, it is still possible to request multiple primary keys (--> we crash) or overlapping constraints.